### DISCUSSION TITLE
Current version

### DISCUSSION THREAD

**Discussion by olibclarke** [zhonge/cryodrgn#158] [Q&A]
Hi,

The current version on the main page is listed as 1.1, but the version reported upon a fresh install is "cryoDRGN 1.0.1.dev126+g04c4a90". Also, the function `cryodrgn select_random` mentioned in the cryodrgn2 quickstart guide doesn't appear to exist, although `cryodrgn abinit_het` does, which I thought only existed in cryodrgn2. I'm wondering if I've done something wrong during installation, or if this is expected?

EDIT: ah - now I see that it should be `cryodrgn_utils select_random`, the cryogrn2 guide must have a typo (or maybe in an unreleased version the syntax has changed?)

Cheers
Oli

**Comment by zhonge** [comment]
Thanks, Oli! I fixed the cryodrgn2 docs to say `cryodrgn_utils select_random`. @vineetbansal can you take a look at the released versions?

**Reply by vineetbansal** [reply]
@olibclarke - the version number you're seeing is telling you that instead of installing a "released" version of cryoDRGN (which you would do by going to the "releases" page, downloading the zip, and then doing a `pip install .` or `python setup.py install` as used to be the case), you checked out the code from some branch (even perhaps `master`) that has changes on top of a release `1.0.1` made in the past. The latest version I see if you were to follow the former procedure is `1.1.0`.

@zhonge - If the release made by us on July 15 2022 is version `1.1.0`, then the current state of the code cannot be `1.1.0` too. This is exactly the benefit of using `setuptools-scm` like we're doing. The git tag **is** the version number going forward. Of course we can create/push a tag or do a GitHub release (same thing) at any point. I'll discuss with you today about who should be making that decision and when.

**Comment by olibclarke** [comment]
@vineetbansal @zhonge thank you very much for the explanation! I followed the install instructions here: https://zhonge.github.io/cryodrgn/pages/installation.html

These instructions included: `git clone https://github.com/zhonge/cryodrgn.git` (which I presume is master).

I see now that these are different from the instructions on the main github docs page. Which one is preferred?

Cheers
Oli

**Comment by olibclarke** [comment]
update - when I checkout version 1.1.0 as suggested on the main github page, job types associated with cryodrgn2 (abinit_homo & abinit_head) do not seem to be included - so I guess the default version is actually later, despite having an earlier version number?

**Reply by vineetbansal** [reply]
@olibclarke - yes, it would appear so. We've put in the machinery to automatically generate/bump version numbers in `cryoDRGN`, but haven't decided what a "good" candidate for a `1.1.1` release would be. @zhonge - is it okay if I tag the current `HEAD` on `master` as version `1.1.1`? I'm also going to remove the mention of:

```
git checkout 1.1.0 # or latest version
```
in the installation README since it will/has already become dated. Users checking out the latest and greatest code can do a `git clone` + `pip install .`. Users needing stability can do a `pip install cryodrgn=1.1.1` (once we add the one extra step in our CI to release the package to pyPI).
